### DISCUSSION TITLE
Grid layout for the ComboBox list

### DISCUSSION THREAD

**Discussion by fregante** [adobe/react-spectrum#3177] [Q&A]
Is it possible to place the items of a ComboBox in a grid instead of the standard list? 

I'm trying to build an icon picker but the icons are displayed in a list, which isn't great. Current code:

```jsx
      <ComboBox defaultItems={iconOptions}>
        {(item) => 
          <Item>
            <Icon
              icon={item.value.id}
              size={32}
            />
          </Item>
        }
      </ComboBox>
```

<img width="252" alt="Screen Shot" src="https://user-images.githubusercontent.com/1402241/171162944-a2d78199-78ed-424c-ae24-010f7251b562.png">

**Comment by snowystinger** [comment]
They should probably have their name next to them as text to make it more accessible, in which case, a list makes sense.
Otherwise, it's difficult to tell what part of what you typed has matched the currently displayed icons, especially if it's a fuzzy match.

**Reply by LFDanLu** [reply]
+1 to what @snowystinger said above. That being said, the grid popup case exists in the [WAI-ARIA 1.2 combobox examples](https://w3c.github.io/aria-practices/examples/combobox/grid-combo.html) but that interaction pattern isn't supported by our combobox hooks. You'd have to modify the keyboard event handlers in those hooks and support a grid popover instead of using a listbox

**Reply by snowystinger** [reply]
I'm not sure that's a great experience, arrow key navigation becomes unreliable/unpredictable. Should left and right arrows navigate the grid cells? or should they move in the input?
I did not try that with VO, I'm not sure what it would say if it's updating both the cell and the cursor location at the same time.
